### PR TITLE
Remove SSR and server-side API requests from web

### LINKED ISSUE #340: [Feature]: Running Immich behind a Reverse-Proxy configured with a Self-signed-certificate
### Feature detail

I run a standard Nginx-Reverse-Proxy with a self-signed-certificate.
So I can access the service with a subdomain like `https://immich.example.local`
If I set `VITE_SERVER_ENDPOINT=https://immich.example.local/api` the Web-Container complains, that it can't reach the resource because of the self-signed-certificate.
If I set `VITE_SERVER_ENDPOINT=http://immich-proxy:2283/api` I get messages in the developer-console saying: `Blocked loading mixed active content “http://immich-proxy:2283/api/asset"`.

Would be great if Immich could be run behind a custom reverse-proxy.
Running latest docker on Ubuntu 20.04.
And thanks for the great work on this one!
Seems that Immich could be the missing piece in the selfhosting photo-puzzle.

### Platform

Web

### PR DESCRIPTION
**Pull request by matthinc**
Fixes #340

### CODE REVIEW

**Comment by zoodyy** [comment]
This is great! When the web app is build statically, we can remove the web docker container and simply serve the static html/js/css files via nginx. This simplifies the setup (esp. solves the VITE_SERVER_ENDPOINT problem) and should further improve immich web performance on low-end servers

File: web/src/lib/user_auth.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { api, UserResponseDto } from '@api';
+import { goto } from '$app/navigation';
+import { get } from 'svelte/store';
+import { session } from '$app/stores';
+
+export async function checkUserAuthStatus(): Promise<UserResponseDto>  {
+    const user = get(session).user;
```
Comment (alextran1502) [inline web/src/lib/user_auth.ts:7]: You can use `$session` to get the latest value of session `$` is for reactive in Svelte

Ref: https://svelte.dev/tutorial/auto-subscriptions

**Comment by alextran1502** [comment]
What is your testing plan to ensure all of the removed SSR requests that have been moved to the client side are still working?

File: web/src/lib/user_auth.ts (hunk 1 above)
Comment (matthinc) [reply web/src/lib/user_auth.ts:7]: That seems to work only in .svelte files and not in .ts.

**Comment by matthinc** [comment]
> What is your testing plan to ensure all of the removed SSR requests that have been moved to the client side are still working?

I'm trying to work component by component and test each functionality independently and then a lot of manual testing before merge. But yeah, it's a challenge. I'd love to discuss UI tests.

File: web/src/lib/user_auth.ts (hunk 1 above)
Comment (alextran1502) [reply web/src/lib/user_auth.ts:7]: Ah that is correct

**Comment by alextran1502** [comment]
> > What is your testing plan to ensure all of the removed SSR requests that have been moved to the client side are still working?
> 
> I'm trying to work component by component and test each functionality independently and then a lot of manual testing before merge. But yeah, it's a challenge. I'd love to discuss UI tests.

For UI testing, I am thinking about Cypress, I have been looking at it for the past few days, and I will probably try to incorporate it after I finish the Album PR. I thought we would remove the `immich-web` container, but I realized I messed up with the idea of making the web fully SPA which I don't think we are going to. So everything you see in development will translate to production. (For the fully SPA mode, the built project might be different from development since it bundles everything into a single JS file)

**Comment by alextran1502** [comment]
Supersede by #377

**Pull request closed without merging**